src/docfx/lib/log/DocfxException.cs

```diff
@@ -21,9 +21,18 @@ public DocfxException(Error error, Exception innerException = null)
 
         public static bool IsDocfxException(Exception ex, out DocfxException docfxException)
         {
-            docfxException = ex as DocfxException;
+            while (ex != null)
+            {
+                if (ex is DocfxException de)
+                {
+                    docfxException = de;
+                    return true;
+                }
+                ex = ex.InnerException;
```

Comment:
would we avoid change the input exception?

Reply:
it is a recursive algorithm so I need to change it, do you suggest having a new variable?